server/server.go

```diff
@@ -91,12 +91,27 @@ func httpHandler(ax *flusher.Axiom, runtimeDone chan struct{}) http.HandlerFunc
 			e["_time"], e["time"] = e["time"], nil
 
 			if e["type"] == "function" {
-				// parse the record
-				matches := logLineRgx.FindStringSubmatch(e["record"].(string))
-				if len(matches) == 5 {
-					e["message"] = matches[4]
-					e["record"] = map[string]any{"requestId": matches[2], "message": e["record"], "timestamp": matches[1]}
-					e["level"] = strings.ToLower(matches[3])
+				e["message"] = e["record"]
+				if recordStr, ok := e["record"].(string); ok {
+					recordStr = strings.Trim(recordStr, "\n")
+					// parse the record
+					// first check if the record is a json object, if not parse it as a text log line
+					if recordStr[0] == '{' && recordStr[len(recordStr)-1] == '}' {
```

Comment:
This panics if the string is empty, please check first

Reply:
great catch. fixed.